src/lib/expression/expression_functional.hpp

```diff
@@ -0,0 +1,225 @@
+#pragma once
+
+#include <memory>
+
+#include "abstract_expression.hpp"
+#include "aggregate_expression.hpp"
+#include "arithmetic_expression.hpp"
+#include "between_expression.hpp"
+#include "binary_predicate_expression.hpp"
+#include "case_expression.hpp"
+#include "cast_expression.hpp"
+#include "exists_expression.hpp"
+#include "extract_expression.hpp"
+#include "function_expression.hpp"
+#include "in_expression.hpp"
+#include "is_null_expression.hpp"
+#include "list_expression.hpp"
+#include "logical_expression.hpp"
+#include "lqp_column_expression.hpp"
+#include "lqp_select_expression.hpp"
+#include "parameter_expression.hpp"
+#include "pqp_select_expression.hpp"
+#include "unary_minus_expression.hpp"
+#include "value_expression.hpp"
+
+/**
+ * This file provides convenience methods to create (nested) Expression objects with little boilerplate.
+ *
+ * NOTE: functions suffixed with "_" (e.g. equals_()) to alert the unsuspecting reader to the fact this is something
+ *       different thant the equality check he might expect when reading "equals" *
+ *
+ * In Hyrise we say...
+ *      case_(equals_(a, 123),
+ *            b,
+ *            case_(equals_(a, 1234),
+ *                  a,
+ *                  null_()))
+ *
+ * ...and it actually means:
+ *     const auto value_123 = std::make_shared<ValueExpression>(123);
+ *     const auto value_1234 = std::make_shared<ValueExpression>(1234);
+ *     const auto a_eq_123 = std::make_shared<BinaryPredicateExpression>(PredicateCondition::Equals, int_float_a_expression, value_123);
+ *     const auto a_eq_1234 = std::make_shared<BinaryPredicateExpression>(PredicateCondition::Equals, int_float_a_expression, value_1234);
+ *     const auto null_value = std::make_shared<ValueExpression>(NullValue{});
+ *     const auto case_a_eq_1234 = std::make_shared<CaseExpression>(a_eq_1234, int_float_a_expression, null_value);
+ *     const auto case_a_eq_123 = std::make_shared<CaseExpression>(a_eq_123, int_float_b_expression, case_a_eq_1234);
+ *
+ *  ...and I think that's beautiful.
+ */
+
+namespace opossum {
+
+class AbstractOperator;
+class LQPColumnReference;
+
+namespace expression_functional {
```

Comment:
Added a comment
```
/**
 * expression_"functional", since it supplies a functional-programming like interface to build nested expressions
 */
```
not sure that makes it better?

Reply:
Works for me.